samza-api/src/main/java/org/apache/samza/operators/windows/Windows.java

```diff
@@ -16,85 +16,378 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
 package org.apache.samza.operators.windows;
 
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.operators.MessageStream;
 import org.apache.samza.operators.data.MessageEnvelope;
+import org.apache.samza.operators.triggers.Duration;
+import org.apache.samza.operators.triggers.TimeTrigger;
+import org.apache.samza.operators.triggers.Trigger;
+import org.apache.samza.operators.triggers.Triggers;
 
 import java.util.Collection;
+import java.util.function.BiFunction;
 import java.util.function.Function;
 
-
 /**
- * This class defines a collection of {@link Window} functions. The public classes and methods here are intended to be
- * used by the user (i.e. programmers) to create {@link Window} function directly.
+ * Factory methods for creating a {@link WindowFunction}. Provides APIs for specifying different types of {@link WindowFunction}s.
```

Comment:
-It's not obvious how the type inference would work if you had a Builder - Esp, if you have a setAggregatorFunction() that modifies the type of the `Builder`. 

-An _identity fold aggregator_ is always defined for a Window. If users don't specify an aggregator, we'll use the default one. So, maybe aggregator makes sense as an immutable constructor parameter.